src/Libraries/CoreNodeModels/Equals.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dynamo.Graph.Nodes;
+using Newtonsoft.Json;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
+
+namespace CoreNodeModels
+{
+    [NodeName("==")]
+    [NodeDescription("Equals with tolerance input")]
+    [NodeCategory("Core.Math")]
+    [NodeSearchTags("Equals")]
+    [InPortTypes("double ", "double", "double")]
+    [OutPortTypes("bool")]
+    [IsDesignScriptCompatible]
+    public class Equals : NodeModel
+    {
+        private static readonly string functionName = nameof(DSCore.List.Equals);
+
+        [JsonConstructor]
+        private Equals(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
+        {
+        }
+
+        public Equals()
+        {
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("lhs", "integer or double value")));
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("rhs", "integer or double value")));
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("tolerance", "tolerance = 0.00001")));
+            OutPorts.Add(new PortModel(PortType.Output, this, new PortData("bool", "result of equality check")));
+            RegisterAllPorts();
+        }
+
+        public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
```

Comment:
does this deal with replication /lacing? or do you need to add some special AST for that as well?

Reply:
@aparajit-pratap 
UseLevelAndReplicationGuide()

and it seems this is missing from a bunch of nodeModel nodes .... 😢

Reply:
Yeah I think I had created a task once upon a time to add these to other nodes like the Range node.

Reply:
https://jira.autodesk.com/browse/DYN-1034

Reply:
Hmm, in the case of the Range node, the option is enabled but doesn't work due to a missing `UseLevelAndReplicationGuide` I'm guessing.

Reply:
Further investigation reveals that the Range UI node compiles down to the range expression in DS and since replication guides don't work for the range expression (filed as a separate bug), enabling the `UseLevelAndReplicationGuide` may not suffice.